packages/grafana-data/src/vector/ArrayVector.ts

```diff
@@ -1,47 +1,34 @@
-import { FunctionalVector } from './FunctionalVector';
-
 /**
  * @public
+ *
+ * @deprecated use a simple Array<T>
  */
-export class ArrayVector<T = any> extends FunctionalVector<T> {
-  buffer: T[];
-
-  // This constructor needs to accept any otherwise the value takes over the definition
-  constructor(buffer?: any[]) {
-    super();
-    this.buffer = buffer ? buffer : [];
+export class ArrayVector<T = any> extends Array<T> {
+  get buffer() {
+    return this;
   }
 
-  get length() {
-    return this.buffer.length;
-  }
+  set buffer(values: any[]) {
+    this.length = 0;
 
-  add(value: T): void {
-    this.buffer.push(value);
-  }
+    const len = values?.length;
 
-  get(index: number): T {
-    return this.buffer[index];
-  }
+    if (len) {
+      let chonkSize = 65e3;
+      let numChonks = Math.ceil(len / chonkSize);
 
-  set(index: number, value: T) {
-    this.buffer[index] = value;
+      for (let chonkIdx = 0; chonkIdx < numChonks; chonkIdx++) {
+        this.push.apply(this, values.slice(chonkIdx * chonkSize, (chonkIdx + 1) * chonkSize));
+      }
+    }
   }
 
-  /** support standard array push syntax */
-  push(...vals: T[]): number {
-    return this.buffer.push(...vals);
-  }
-
-  reverse() {
-    return this.buffer.reverse();
-  }
-
-  toArray(): T[] {
-    return this.buffer;
+  constructor(buffer?: T[]) {
+    super();
+    this.buffer = buffer ?? [];
```

Comment:
this now shallow-copies the passed-in buffer rather than absorbing it by reference. since this class now extends Array, we can't avoid this unfortunately. this new behavior required fixes in two places which had these patterns:

1. pre-allocate a buffer via `let buf = new Array(10)`
2. wrap in with `field.values = ArrayVector(buf)`
3. mutate the buffer expecting it to reflect in field.values via `buf[i] = 100`

<br>

1. Create a new mutable frame: `let mutFrame = new MutableDataFrame()`;
2. Create some field: `let field = {name: 'a', values: new ArrayVector()}`
3. Add field: `mutFrame.addField(field)` (this shallow copies field internally, and wraps .values buffers in ArrayVector)
4. Try to add values to the field, expecting the values to change in the mutable frame field's copy: `field.values.add("a")`